src/uu/more/src/more.rs

```diff
@@ -10,74 +10,144 @@
 #[macro_use]
 extern crate uucore;
 
-use std::fs::File;
-use std::io::{stdin, stdout, BufRead, BufReader, Read, Write};
+use std::{
+    convert::TryInto,
+    fs::File,
+    io::{stdout, BufReader, Read, Write},
+    path::Path,
+    time::Duration,
+};
 
 #[cfg(all(unix, not(target_os = "fuchsia")))]
 extern crate nix;
-#[cfg(all(unix, not(target_os = "fuchsia")))]
-use nix::sys::termios::{self, LocalFlags, SetArg};
 
 #[cfg(target_os = "redox")]
 extern crate redox_termios;
 #[cfg(target_os = "redox")]
 extern crate syscall;
 
-use clap::{App, Arg, ArgMatches};
-
-static VERSION: &str = env!("CARGO_PKG_VERSION");
-static ABOUT: &str = "A file perusal filter for CRT viewing.";
-
-mod options {
-    pub const FILE: &str = "file";
-}
-
-fn get_usage() -> String {
-    format!("{} [options] <file>...", executable!())
-}
+use clap::{App, Arg};
+use crossterm::{
+    event::{self, Event, KeyCode, KeyEvent, KeyModifiers},
+    execute, terminal,
+};
 
-pub fn uumain(args: impl uucore::Args) -> i32 {
-    let usage = get_usage();
+pub fn uumain(_args: impl uucore::Args) -> i32 {
+    let matches = build_app().get_matches();
 
-    let matches = App::new(executable!())
-        .version(VERSION)
-        .usage(usage.as_str())
-        .about(ABOUT)
-        .arg(
-            Arg::with_name(options::FILE)
-                .number_of_values(1)
-                .multiple(true),
-        )
-        .get_matches_from(args);
+    if let Some(filenames) = matches.values_of("files") {
+        let mut buff = String::new();
+        for fname in filenames {
+            if Path::new(fname).is_dir() {
+                show_usage_error!("'{}' is a directory.", fname);
+            }
+            let mut reader = BufReader::new(File::open(fname).unwrap());
+            reader.read_to_string(&mut buff).unwrap();
+            more(&buff);
+        }
+    }
 
     // FixME: fail without panic for now; but `more` should work with no arguments (ie, for piped input)
+    /*
     if let None | Some("-") = matches.value_of(options::FILE) {
         show_usage_error!("Reading from stdin isn't supported yet.");
         return 1;
     }
-
-    if let Some(x) = matches.value_of(options::FILE) {
-        let path = std::path::Path::new(x);
-        if path.is_dir() {
-            show_usage_error!("'{}' is a directory.", x);
-            return 1;
-        }
-    }
-
-    more(matches);
-
+    */
     0
 }
 
+fn build_app() -> App<'static, 'static> {
+    App::new(executable!())
+        .about("A file perusal filter for CRT viewing.")
+        .version(env!("CARGO_PKG_VERSION"))
+        .arg(
+            Arg::with_name("silent")
```

Comment:
please uses options like in other binaries:
https://github.com/uutils/coreutils/search?q=%22pub+mod+options%22